conans/test/integration/layout/test_in_cache.py

```diff
@@ -0,0 +1,114 @@
+import os
+import pytest
+
+from conans import load
+from conans.model.ref import ConanFileReference, PackageReference
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient, NO_SETTINGS_PACKAGE_ID
+
+
+@pytest.fixture
+def conanfile():
+    conan_file = str(GenConanfile().with_import("from conans import tools").with_import("import os").
+                     with_require("base/1.0"))
+
+    # FIXME: The configure is not valid to change the layout, we need the settings and options
+    #        ready
+    conan_file += """
+    no_copy_sources = True
+
+    def configure(self):
+        self.layout.source.folder = "my_sources"
+        self.layout.build.folder = "my_build"
+
+    def source(self):
+        self.output.warn("Source folder: {}".format(self.source_folder))
+        tools.save("source.h", "foo")
+
+    def build(self):
+        self.output.warn("Build folder: {}".format(self.build_folder))
+        tools.save("build.lib", "bar")
+
+    def package(self):
+        self.output.warn("Package folder: {}".format(self.package_folder))
+        tools.save(os.path.join(self.package_folder, "LICENSE"), "bar")
+        self.copy("*.h", dst="include")
+        self.copy("*.lib", dst="lib")
+
+    def package_info(self):
+        # This will be easier when the layout declares also the includedirs etc
+        self.cpp_info.includedirs = [os.path.join(self.package_folder, "include")]
```

Comment:
Not sure about this. The ``includedirs`` was always relative to the package_folder, can't we keep it without this?

Reply:
Sure, now that there is no `package` layout, that is not necessary.